### ISSUE TITLE
SET5验证集PSNR只有2.7

### ISSUE THREAD

**Issue by hahaVVV402** [XPixelGroup/ClassSR#62]
运行python train.py -opt options/train/train_FSRCNN.yml在验证时psnr只有2.7
![Image](https://github.com/user-attachments/assets/baf27aa7-8cca-4e69-ae97-fb82d1acafb5)
这是运行的配置文件
#### general settings
name: FSRCNN_branch1
use_tb_logger: true
model: sr
distortion: sr
scale: 4
gpu_ids: [0]

#### network structures
network_G:
  which_model_G: fsrcnn
  in_nc: 3
  out_nc: 3
  upscale: 4
  d: 16
  s: 12
  m: 4

#### datasets
datasets:
  train:
    name: DIV2K
    mode: LQGT
    dataroot_GT: D:\\Coding\\ClassSR\\codes\data\\DIV2K_scale_sub_psnr_GT_class1
    dataroot_LQ: D:\\Coding\\ClassSR\\codes\data\\DIV2K_scale_sub_psnr_LR_class1

    use_shuffle: true
    n_workers: 3  # per GPU
    batch_size: 16
    GT_size: 128
    use_flip: true
    use_rot: true
    color: RGB
  val:
    name: set5
    mode: LQGT
    dataroot_GT: D:\\Coding\\ClassSR\\codes\\data\\Set5\\HR
    dataroot_LQ: D:\\Coding\\ClassSR\\codes\data\\Set5\LR

#### path
path:
  pretrain_model_G: ~
  strict_load: true
  resume_state: ~

#### training settings: learning rate scheme, loss
train:
  lr_G: !!float 1e-3
  lr_scheme: CosineAnnealingLR_Restart
  beta1: 0.9
  beta2: 0.99
  niter: 500000
  warmup_iter: -1  # no warm up
  T_period: [500000]
  restarts: [500000]
  restart_weights: [1]
  eta_min: !!float 1e-7

  pixel_criterion: l1
  pixel_weight: 1.0

  manual_seed: 10
  val_freq: !!float 5000

#### logger
logger:
  print_freq: 100
  save_checkpoint_freq: !!float 5000
请问这是什么原因

**Comment by Xiangtaokong** [comment]
e+n是科学计数法10的n次方，2.78e+01 意思是 27.8。

**Comment by hahaVVV402** [comment]
感谢回答
使用python train_ClassSR.py -opt options/train/train_ClassSR_FSRCNN.yml 训练时 在val阶段会报错

![Image](https://github.com/user-attachments/assets/99286db9-dd93-4694-82fb-319581f18ed9)
查阅以前issue看到回答说可能与数据集有关
我的val数据集是使用python extract_subimages_test.py DIV2K_valid, index 801-810 制作成的 这样有问题吗？
这是训练的配置文件
#### general settings
name: ClassSR_FSRCNN
use_tb_logger: true
model: ClassSR
distortion: sr
scale: 4
gpu_ids: [0]
patch_size: 32
step: 28
fix_SR_module: true



#### datasets
datasets:
  train:
    name: DIV2K
    mode: LQGT
    dataroot_GT: D:\Coding\ClassSR\codes\data\DIV2K_scale_sub\GT
    dataroot_LQ: D:\Coding\ClassSR\codes\data\DIV2K_scale_sub\LR


    use_shuffle: true
    n_workers: 6  # per GPU
    batch_size: 32
    GT_size: 128
    use_flip: true
    use_rot: true
    color: RGB
  val:
    name: DIV2K10
    mode: LQGT
    dataroot_GT: D:\Coding\ClassSR\codes\data\DIV2K_valid_HR_sub\GT
    dataroot_LQ: D:\Coding\ClassSR\codes\data\DIV2K_valid_HR_sub\LR

#### network structures
network_G:
  which_model_G: classSR_3class_fsrcnn_net
  in_nc: 3
  out_nc: 3

#### path
path:
  pretrain_model_G: ~
  pretrain_model_classifier: ~
  pretrain_model_G_branch1: D:\Coding\ClassSR\experiments\pre_trained_models\FSRCNN_branch1.pth
  pretrain_model_G_branch2: D:\Coding\ClassSR\experiments\pre_trained_models\FSRCNN_branch2.pth
  pretrain_model_G_branch3: D:\Coding\ClassSR\experiments\pre_trained_models\FSRCNN_branch3.pth
  resume_state: ~
  strict_load: true

#### training settings: learning rate scheme, loss
train:
  lr_G: !!float 2e-4
  lr_scheme: CosineAnnealingLR_Restart
  beta1: 0.9
  beta2: 0.99
  niter: 200000
  warmup_iter: -1  # no warm up
  T_period: [500000, 500000]
  restarts: [500000]
  restart_weights: [1]
  eta_min: !!float 1e-7

  pixel_criterion: ClassSR_loss
  l1w: 1000
  class_loss_w: 0.5
  average_loss_w: 3

  manual_seed: 10
  val_freq: !!float 1000

#### logger
logger:
  print_freq: 100
  save_checkpoint_freq: !!float 2000

**Comment by Xiangtaokong** [comment]
时间久远记不太清了，val数据集好像不用手动切割，直接用整图。可以试一下



发自我的iPhone


------------------ 原始邮件 ------------------
发件人: WENWU JIANG ***@***.***&gt;
发送时间: 2025年3月21日 19:41
收件人: XPixelGroup/ClassSR ***@***.***&gt;
抄送: Xiangtao Kong ***@***.***&gt;, Comment ***@***.***&gt;
主题: Re: [XPixelGroup/ClassSR] SET5验证集PSNR只有2.7 (Issue #62)

**Comment by hahaVVV402** [comment]
感谢回答，使用整图后不报错了

**Comment by hahaVVV402** [comment]
使用python train_ClassSR.py -opt options/train/train_ClassSR_CARN.yml 训练ClassSR_CARN 为什么输出日志只有验证过程没有训练过程 对比了ClassSR_FSNR训练配置相同但日志有训练过程

![Image](https://github.com/user-attachments/assets/7a3050e1-25bd-4a48-aec7-2519d532072f)

#### general settings
name: ClassSR_CARN
use_tb_logger: true
model: ClassSR
distortion: sr
scale: 4
gpu_ids: [0]
patch_size: 32
step: 28
fix_SR_module: true



#### datasets
datasets:
  train:
    name: DIV2K
    mode: LQGT
    dataroot_GT: D:\Coding\ClassSR\codes\data\DIV2K_scale_sub\GT
    dataroot_LQ: D:\Coding\ClassSR\codes\data\DIV2K_scale_sub\LR


    use_shuffle: true
    n_workers: 6  # per GPU
    batch_size: 96
    GT_size: 128
    use_flip: true
    use_rot: true
    color: RGB
  val:
    name: DIV2K10
    mode: LQGT
    dataroot_GT: D:\Coding\ClassSR\codes\data\DIV2K_valid_10\GT
    dataroot_LQ: D:\Coding\ClassSR\codes\data\DIV2K_valid_10\LR


#### network structures
network_G:
  which_model_G: classSR_3class_carn
  in_nc: 3
  out_nc: 3

#### path
path:
  pretrain_model_G: ~
  pretrain_model_classifier: ~
  pretrain_model_G_branch1: D:\Coding\ClassSR\experiments\pre_trained_models\CARN_branch1.pth
  pretrain_model_G_branch2: D:\Coding\ClassSR\experiments\pre_trained_models\CARN_branch2.pth
  pretrain_model_G_branch3: D:\Coding\ClassSR\experiments\pre_trained_models\CARN_branch3.pth
  resume_state: ~
  strict_load: true

#### training settings: learning rate scheme, loss
train:
  lr_G: !!float 2e-4
  lr_scheme: CosineAnnealingLR_Restart
  beta1: 0.9
  beta2: 0.99
  niter: 200000
  warmup_iter: -1  # no warm up
  T_period: [500000, 500000]
  restarts: [500000]
  restart_weights: [0.05]
  eta_min: !!float 1e-7

  pixel_criterion: ClassSR_loss
  l1w: 1000
  class_loss_w: 0.5
  average_loss_w: 3


  manual_seed: 10
  val_freq: !!float 1

#### logger
logger:
  print_freq: 100
  save_checkpoint_freq: !!float 2000
#### general settings
name: ClassSR_FSRCNN
use_tb_logger: true
model: ClassSR
distortion: sr
scale: 4
gpu_ids: [0]
patch_size: 32
step: 28
fix_SR_module: true



#### datasets
datasets:
  train:
    name: DIV2K
    mode: LQGT
    dataroot_GT: D:\Coding\ClassSR\codes\data\DIV2K_scale_sub\GT
    dataroot_LQ: D:\Coding\ClassSR\codes\data\DIV2K_scale_sub\LR


    use_shuffle: true
    n_workers: 6  # per GPU
    batch_size: 32
    GT_size: 128
    use_flip: true
    use_rot: true
    color: RGB
  val:
    name: DIV2K10
    mode: LQGT
    dataroot_GT: D:\Coding\ClassSR\codes\data\DIV2K_valid_10\GT
    dataroot_LQ: D:\Coding\ClassSR\codes\data\DIV2K_valid_10\LR

#### network structures
network_G:
  which_model_G: classSR_3class_fsrcnn_net
  in_nc: 3
  out_nc: 3

#### path
path:
  pretrain_model_G: ~
  pretrain_model_classifier: ~
  pretrain_model_G_branch1: D:\Coding\ClassSR\experiments\pre_trained_models\FSRCNN_branch1.pth
  pretrain_model_G_branch2: D:\Coding\ClassSR\experiments\pre_trained_models\FSRCNN_branch2.pth
  pretrain_model_G_branch3: D:\Coding\ClassSR\experiments\pre_trained_models\FSRCNN_branch3.pth
  resume_state: ~
  strict_load: true

#### training settings: learning rate scheme, loss
train:
  lr_G: !!float 2e-4
  lr_scheme: CosineAnnealingLR_Restart
  beta1: 0.9
  beta2: 0.99
  niter: 200000
  warmup_iter: -1  # no warm up
  T_period: [500000, 500000]
  restarts: [500000]
  restart_weights: [1]
  eta_min: !!float 1e-7

  pixel_criterion: ClassSR_loss
  l1w: 1000
  class_loss_w: 0.5
  average_loss_w: 3

  manual_seed: 10
  val_freq: !!float 1000

#### logger
logger:
  print_freq: 100
  save_checkpoint_freq: !!float 2000

**Comment by Xiangtaokong** [comment]
val_freq: !!float 1 是验证的频率，1就是每个训练循环都要验证，设大一点就可以了。

**Comment by hahaVVV402** [comment]
感谢回答

**Comment by hahaVVV402** [comment]
抱歉再次打扰还有两个问题想请教一下
1.请问cal_FLOPs函数中每个分支的FLOPs值时如何确定的 我使用thop库计算的FLOPs为

![Image](https://github.com/user-attachments/assets/40979c4b-86c2-41fb-99fc-c32bb205b588)

![Image](https://github.com/user-attachments/assets/2fb9785c-0cc9-42dc-a4e1-afccc2926e96)
2.固定SR_model即“fix_SR_module: true”训练完成后是将训练完保存的参数在pretrain_model_G:加载 然后开始最后一阶段的训练吗 第三阶段的训练是不是不需要导入三个分支的预训练权重 或者说正确做法是什么？

![Image](https://github.com/user-attachments/assets/8d819a61-e122-45f6-a93f-2108738e5035)
